File: models/session.py
from __future__ import annotations
from typing import Optional
from pydantic import BaseModel, Field, ConfigDict


class SessionPushRequest(BaseModel):
    """
    Push session metadata to Make.com → svlbh-v2.
    Key format: programCode-patientId-sessionNum-practitionerCode (e.g. 00-12-002-0301)
    """
    model_config = ConfigDict(populate_by_name=True)

    session_key: str = Field(..., alias="sessionKey")
    patient_id: str = Field(..., alias="patientId")
    session_num: str = Field(..., alias="sessionNum")
    program_code: str = Field(..., alias="programCode")
    practitioner_code: str = Field(..., alias="practitionerCode")
    therapist_name: Optional[str] = Field(None, alias="therapistName")
    status: str = Field("ACTIVE")
    event_count: int = Field(0, alias="eventCount")
    liberated_count: int = Field(0, alias="liberatedCount")
    platform: str = Field("android")


class SessionPullRequest(BaseModel):
    model_config = ConfigDict(populate_by_name=True)

    session_key: str = Field(..., alias="sessionKey")


class SessionPullResponse(BaseModel):
    session_key: str
    patient_id: Optional[str] = None
    session_num: Optional[str] = None
    program_code: Optional[str] = None
    practitioner_code: Optional[str] = None
    therapist_name: Optional[str] = None
    status: Optional[str] = None
    event_count: Optional[int] = None
    liberated_count: Optional[int] = None
    found: bool = False
    timestamp: Optional[int] = None

File: android/vlbh-energy-sdk/src/main/java/energy/vlbh/sdk/models/Session.kt
package energy.vlbh.sdk.models

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class SessionPushRequest(
    val sessionKey: String,
    val patientId: String,
    val sessionNum: String,
    val programCode: String,
    val practitionerCode: String,
    val therapistName: String? = null,
    val status: String = "ACTIVE",
    val eventCount: Int = 0,
    val liberatedCount: Int = 0,
    val platform: String = "android"
)

@Serializable
data class SessionPullRequest(
    val sessionKey: String
)

@Serializable
data class SessionPullResponse(
    @SerialName("session_key") val sessionKey: String,
    @SerialName("patient_id") val patientId: String? = null,
    @SerialName("session_num") val sessionNum: String? = null,
    @SerialName("program_code") val programCode: String? = null,
    @SerialName("practitioner_code") val practitionerCode: String? = null,
    @SerialName("therapist_name") val therapistName: String? = null,
    val status: String? = null,
    @SerialName("event_count") val eventCount: Int? = null,
    @SerialName("liberated_count") val liberatedCount: Int? = null,
    val found: Boolean,
    val timestamp: Long? = null
)
